subsys/usb/usb_c/usbc_pe_common_internal.h

```diff
@@ -64,6 +64,9 @@ enum usbc_pe_state {
 	/** PE_Suspend. Not part of the PD specification. */
 	PE_SUSPEND,
 
+	/** PE_SENDER_RESPONSE. Not part of the PD specification. */
+	PE_SENDER_RESPONSE,
```

Comment:
I think this would add complexity running a state machine composed of states from multiple smf_state arrays. I added comments that explain the uses of the parent state.

Reply:
What happens to `ctx.current` when the parent state is run?  If the current state still points to the child state, then creating two arrays shouldn't impact any of the other logic.

The main drawback to putting the parent state into `pe_states[]` is that the assert in `pe_set_state()` fails to catch an error of trying to directly enter the parent state: `pe_set_state(dev, PE_SENDER_RESPONSE_PARENT);`.

@DianaZig, what do you think?

Reply:
@keith-zephyr This is true for all other super states. TC_CC_OPEN_SUPER_STATE and TC_CC_RD_SUPER_STATE. I don't disagree with your argument but for consistency, we should keep the parent and sibling states in the same array and make your suggested change to all state machines in a separate PR.